src/parallel.cpp

```diff
@@ -28,8 +28,9 @@ using namespace std;
 
 void AddScriptCheckThreads(int i, CCheckQueue<CScriptCheck>* pqueue)
 {
-    string tName = "bitcoin-scriptchk" + i;
-    RenameThread(tName.c_str());
+    ostringstream tName;
+    tName << "bitcoin-scriptchk" << i;
+    RenameThread(tName.str().c_str());
```

Comment:
call me traditional but this looks like lots more internal complexity and several more mallocs and frees than the original code.  What was the problem with it?  

Also, given that the max length of i is known we could be even more efficient: this might be a great opportunity to be efficient and use standard C:
char* s[xxx];
snprintf("bitcoin-scriptchk%d", i);
OR
memcpy and itoa?

RenameThread(s);